### ISSUE TITLE
Can't execute any Modin function (ValueError: tuple is not allowed for map key)

### ISSUE THREAD

**Issue by HiIamJeff** [modin-project/modin#1104]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10 
- **Modin version** (`modin.__version__`): '0.7.0
- **Python version**: 3.7.1 
- **Code we can use to reproduce**:
```
import os
os.environ["MODIN_ENGINE"] = "dask"
import modin.pandas as pd
import numpy as np
df = pd.DataFrame([1,2,3])
```

### Describe the problem

I am a new user who just follows your installation page to install Modin and tried out a little bit. But I found that I couldn't even execute a simple function through your package (ValueError: tuple is not allowed for map key). I have installed packages by the following command.  
```
pip install -U modin # -U for upgrade in case you have an older version
pip install modin[dask] # Install Modin dependencies and Dask to run on Dask
```

p.s. By the way, every time I imported the package (import modin.pandas as pd), the jupyter notebook would give me the following warning. I couldn't really figure out the cause.

```
UserWarning: The Dask Engine for Modin is experimental.
UserWarning: 
Failed to start diagnostics server on port 8787. [WinError 10013] 嘗試存取通訊端被拒絕，因為存取權限不足。 ## This means the access is denied because the accuess authority is insufficient

```
### Source code / logs

```
UserWarning: Distributing <class 'list'> object. This may take some time.
distributed.protocol.core - CRITICAL - Failed to deserialize
Traceback (most recent call last):
  File "C:\Users\ADMIN\Anaconda3\lib\site-packages\distributed\protocol\core.py", line 114, in loads
    header = msgpack.loads(header, use_list=False, **msgpack_opts)
  File "msgpack\_unpacker.pyx", line 195, in msgpack._cmsgpack.unpackb
ValueError: tuple is not allowed for map key
distributed.core - ERROR - tuple is not allowed for map key
Traceback (most recent call last):
  File "C:\Users\ADMIN\Anaconda3\lib\site-packages\distributed\core.py", line 297, in handle_comm
    msg = yield comm.read()
  File "C:\Users\ADMIN\Anaconda3\lib\site-packages\tornado\gen.py", line 1133, in run
    value = future.result()
  File "C:\Users\ADMIN\Anaconda3\lib\site-packages\tornado\gen.py", line 1141, in run
    yielded = self.gen.throw(*exc_info)
  File "C:\Users\ADMIN\Anaconda3\lib\site-packages\distributed\comm\tcp.py", line 206, in read
    deserializers=deserializers)
  File "C:\Users\ADMIN\Anaconda3\lib\site-packages\tornado\gen.py", line 1133, in run
    value = future.result()
  File "C:\Users\ADMIN\Anaconda3\lib\site-packages\tornado\gen.py", line 326, in wrapper
    yielded = next(result)
  File "C:\Users\ADMIN\Anaconda3\lib\site-packages\distributed\comm\utils.py", line 79, in from_frames
    res = _from_frames()
  File "C:\Users\ADMIN\Anaconda3\lib\site-packages\distributed\comm\utils.py", line 65, in _from_frames
    deserializers=deserializers)
  File "C:\Users\ADMIN\Anaconda3\lib\site-packages\distributed\protocol\core.py", line 114, in loads
    header = msgpack.loads(header, use_list=False, **msgpack_opts)
  File "msgpack\_unpacker.pyx", line 195, in msgpack._cmsgpack.unpackb
ValueError: tuple is not allowed for map key
distributed.protocol.core - CRITICAL - Failed to deserialize
Traceback (most recent call last):
  File "C:\Users\ADMIN\Anaconda3\lib\site-packages\distributed\protocol\core.py", line 114, in loads
    header = msgpack.loads(header, use_list=False, **msgpack_opts)
  File "msgpack\_unpacker.pyx", line 195, in msgpack._cmsgpack.unpackb
ValueError: tuple is not allowed for map key
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-1f9265289ca2> in <module>
      4 import modin.pandas as pd
      5 import numpy as np
----> 6 df = pd.DataFrame([1,2,3])
      7 # data_set = pd.read_csv(data_path+'/'+data_file)

~\Anaconda3\lib\site-packages\modin\pandas\dataframe.py in __init__(self, data, index, columns, dtype, copy, query_compiler)
     84                 data=data, index=index, columns=columns, dtype=dtype, copy=copy
     85             )
---> 86             self._query_compiler = from_pandas(pandas_df)._query_compiler
     87         else:
     88             self._query_compiler = query_compiler

~\Anaconda3\lib\site-packages\modin\pandas\utils.py in from_pandas(df)
     21     from .dataframe import DataFrame
     22 
---> 23     return DataFrame(query_compiler=BaseFactory.from_pandas(df))
     24 
     25 

~\Anaconda3\lib\site-packages\modin\data_management\factories.py in from_pandas(cls, df)
     26     @classmethod
     27     def from_pandas(cls, df):
---> 28         return cls._determine_engine()._from_pandas(df)
     29 
     30     @classmethod

~\Anaconda3\lib\site-packages\modin\data_management\factories.py in _from_pandas(cls, df)
     30     @classmethod
     31     def _from_pandas(cls, df):
---> 32         return cls.io_cls.from_pandas(df)
     33 
     34     @classmethod

~\Anaconda3\lib\site-packages\modin\engines\base\io\io.py in from_pandas(cls, df)
     12     @classmethod
     13     def from_pandas(cls, df):
---> 14         return cls.query_compiler_cls.from_pandas(df, cls.frame_cls)
     15 
     16     @classmethod

~\Anaconda3\lib\site-packages\modin\backends\pandas\query_compiler.py in from_pandas(cls, df, data_cls)
     59     @classmethod
     60     def from_pandas(cls, df, data_cls):
---> 61         return cls(data_cls.from_pandas(df))
     62 
     63     index = property(_get_axis(0), _set_axis(0))

~\Anaconda3\lib\site-packages\modin\engines\base\frame\data.py in from_pandas(cls, df)
   1181         new_columns = df.columns
   1182         new_dtypes = df.dtypes
-> 1183         new_frame, new_lengths, new_widths = cls._frame_mgr_cls.from_pandas(df, True)
   1184         return cls(
   1185             new_frame,

~\Anaconda3\lib\site-packages\modin\engines\base\frame\partition_manager.py in from_pandas(cls, df, return_dims)
    307                 for j in range(0, len(df.columns), col_chunksize)
    308             ]
--> 309             for i in range(0, len(df), row_chunksize)
    310         ]
    311         if not return_dims:

~\Anaconda3\lib\site-packages\modin\engines\base\frame\partition_manager.py in <listcomp>(.0)
    307                 for j in range(0, len(df.columns), col_chunksize)
    308             ]
--> 309             for i in range(0, len(df), row_chunksize)
    310         ]
    311         if not return_dims:

~\Anaconda3\lib\site-packages\modin\engines\base\frame\partition_manager.py in <listcomp>(.0)
    305             [
    306                 put_func(df.iloc[i : i + row_chunksize, j : j + col_chunksize].copy())
--> 307                 for j in range(0, len(df.columns), col_chunksize)
    308             ]
    309             for i in range(0, len(df), row_chunksize)

~\Anaconda3\lib\site-packages\modin\engines\dask\pandas_on_dask_futures\frame\partition.py in put(cls, obj)
    135         """
    136         client = _get_global_client()
--> 137         return cls(client.scatter(obj, hash=False))
    138 
    139     @classmethod

~\Anaconda3\lib\site-packages\distributed\client.py in scatter(self, data, workers, broadcast, direct, hash, maxsize, timeout, asynchronous)
   1872                              broadcast=broadcast, direct=direct,
   1873                              local_worker=local_worker, timeout=timeout,
-> 1874                              asynchronous=asynchronous, hash=hash)
   1875 
   1876     @gen.coroutine

~\Anaconda3\lib\site-packages\distributed\client.py in sync(self, func, *args, **kwargs)
    671             return future
    672         else:
--> 673             return sync(self.loop, func, *args, **kwargs)
    674 
    675     def __repr__(self):

~\Anaconda3\lib\site-packages\distributed\utils.py in sync(loop, func, *args, **kwargs)
    275             e.wait(10)
    276     if error[0]:
--> 277         six.reraise(*error[0])
    278     else:
    279         return result[0]

~\Anaconda3\lib\site-packages\six.py in reraise(tp, value, tb)
    691             if value.__traceback__ is not tb:
    692                 raise value.with_traceback(tb)
--> 693             raise value
    694         finally:
    695             value = None

~\Anaconda3\lib\site-packages\distributed\utils.py in f()
    260             if timeout is not None:
    261                 future = gen.with_timeout(timedelta(seconds=timeout), future)
--> 262             result[0] = yield future
    263         except Exception as exc:
    264             error[0] = sys.exc_info()

~\Anaconda3\lib\site-packages\tornado\gen.py in run(self)
   1131 
   1132                     try:
-> 1133                         value = future.result()
   1134                     except Exception:
   1135                         self.had_exception = True

~\Anaconda3\lib\site-packages\tornado\gen.py in run(self)
   1139                     if exc_info is not None:
   1140                         try:
-> 1141                             yielded = self.gen.throw(*exc_info)
   1142                         finally:
   1143                             # Break up a reference to itself

~\Anaconda3\lib\site-packages\distributed\client.py in _scatter(self, data, workers, broadcast, direct, local_worker, timeout, hash)
   1734                                              client=self.id,
   1735                                              broadcast=broadcast,
-> 1736                                              timeout=timeout)
   1737 
   1738         out = {k: Future(k, self, inform=False) for k in data}

~\Anaconda3\lib\site-packages\tornado\gen.py in run(self)
   1131 
   1132                     try:
-> 1133                         value = future.result()
   1134                     except Exception:
   1135                         self.had_exception = True

~\Anaconda3\lib\site-packages\tornado\gen.py in run(self)
   1139                     if exc_info is not None:
   1140                         try:
-> 1141                             yielded = self.gen.throw(*exc_info)
   1142                         finally:
   1143                             # Break up a reference to itself

~\Anaconda3\lib\site-packages\distributed\core.py in send_recv_from_rpc(**kwargs)
    578             try:
    579                 comm = yield self.live_comm()
--> 580                 result = yield send_recv(comm=comm, op=key, **kwargs)
    581             except (RPCClosed, CommClosedError) as e:
    582                 raise e.__class__("%s: while trying to call remote method %r"

~\Anaconda3\lib\site-packages\tornado\gen.py in run(self)
   1131 
   1132                     try:
-> 1133                         value = future.result()
   1134                     except Exception:
   1135                         self.had_exception = True

~\Anaconda3\lib\site-packages\tornado\gen.py in run(self)
   1139                     if exc_info is not None:
   1140                         try:
-> 1141                             yielded = self.gen.throw(*exc_info)
   1142                         finally:
   1143                             # Break up a reference to itself

~\Anaconda3\lib\site-packages\distributed\core.py in send_recv(comm, reply, serializers, deserializers, **kwargs)
    455         yield comm.write(msg, serializers=serializers, on_error='raise')
    456         if reply:
--> 457             response = yield comm.read(deserializers=deserializers)
    458         else:
    459             response = None

~\Anaconda3\lib\site-packages\tornado\gen.py in run(self)
   1131 
   1132                     try:
-> 1133                         value = future.result()
   1134                     except Exception:
   1135                         self.had_exception = True

~\Anaconda3\lib\site-packages\tornado\gen.py in run(self)
   1139                     if exc_info is not None:
   1140                         try:
-> 1141                             yielded = self.gen.throw(*exc_info)
   1142                         finally:
   1143                             # Break up a reference to itself

~\Anaconda3\lib\site-packages\distributed\comm\tcp.py in read(self, deserializers)
    204                 msg = yield from_frames(frames,
    205                                         deserialize=self.deserialize,
--> 206                                         deserializers=deserializers)
    207             except EOFError:
    208                 # Frames possibly garbled or truncated by communication error

~\Anaconda3\lib\site-packages\tornado\gen.py in run(self)
   1131 
   1132                     try:
-> 1133                         value = future.result()
   1134                     except Exception:
   1135                         self.had_exception = True

~\Anaconda3\lib\site-packages\tornado\gen.py in wrapper(*args, **kwargs)
    324                 try:
    325                     orig_stack_contexts = stack_context._state.contexts
--> 326                     yielded = next(result)
    327                     if stack_context._state.contexts is not orig_stack_contexts:
    328                         yielded = _create_future()

~\Anaconda3\lib\site-packages\distributed\comm\utils.py in from_frames(frames, deserialize, deserializers)
     77         res = yield offload(_from_frames)
     78     else:
---> 79         res = _from_frames()
     80 
     81     raise gen.Return(res)

~\Anaconda3\lib\site-packages\distributed\comm\utils.py in _from_frames()
     63             return protocol.loads(frames,
     64                                   deserialize=deserialize,
---> 65                                   deserializers=deserializers)
     66         except EOFError:
     67             if size > 1000:

~\Anaconda3\lib\site-packages\distributed\protocol\core.py in loads(frames, deserialize, deserializers)
    112 
    113         header = frames.pop()
--> 114         header = msgpack.loads(header, use_list=False, **msgpack_opts)
    115         keys = header['keys']
    116         headers = header['headers']

msgpack\_unpacker.pyx in msgpack._cmsgpack.unpackb()

ValueError: tuple is not allowed for map key
```

**Comment by devin-petersohn** [comment]
Hi @HiIamJeff, thanks for the report!

This is related to https://github.com/dask/distributed/issues/3491. The fix is to `pip install msgpack<1.0`.

**Comment by HiIamJeff** [comment]
Thanks Devin. It works like charm! Much appreciate it.
